File: lms-frontend/src/employee/services/employeeStatusService.js
import api from "../../api/axios";

const BASE_URL =
    "/api/employee-statuses";

const getAllEmployeeStatuses = () => {

    return api.get(
        BASE_URL
    );
};

const getEmployeeStatusById = (
    employeeStatusId
) => {

    return api.get(
        `${BASE_URL}/${employeeStatusId}`
    );
};

const createEmployeeStatus = (
    statusData
) => {

    return api.post(
        BASE_URL,
        statusData
    );
};

const updateEmployeeStatus = (
    employeeStatusId,
    statusData
) => {

    return api.put(
        `${BASE_URL}/${employeeStatusId}`,
        statusData
    );
};

const deleteEmployeeStatus = (
    employeeStatusId
) => {

    return api.delete(
        `${BASE_URL}/${employeeStatusId}`
    );
};

const employeeStatusService = {

    getAllEmployeeStatuses,
    getEmployeeStatusById,
    createEmployeeStatus,
    updateEmployeeStatus,
    deleteEmployeeStatus
};

export default employeeStatusService;

File: backend/src/main/java/com/lms_erp/employee/service/EmployeeStatusService.java
package com.lms_erp.employee.service;

import com.lms_erp.employee.dto.EmployeeStatusRequest;
import com.lms_erp.employee.dto.EmployeeStatusResponse;

import java.util.List;

public interface EmployeeStatusService {

    EmployeeStatusResponse createEmployeeStatus(
            EmployeeStatusRequest request);

    EmployeeStatusResponse updateEmployeeStatus(
            Long employeeStatusId,
            EmployeeStatusRequest request);

    EmployeeStatusResponse getEmployeeStatusById(
            Long employeeStatusId);

    List<EmployeeStatusResponse> getAllEmployeeStatuses();

    void deleteEmployeeStatus(Long employeeStatusId);
}